include/deal.II/base/symmetric_tensor.h

```diff
@@ -3795,16 +3925,91 @@ operator * (const SymmetricTensor<2,dim,Number> &src1,
  * @relates SymmetricTensor
  * @author Wolfgang Bangerth, 2005
  */
-template <int dim, typename Number>
-Tensor<1,dim,Number>
-operator * (const Tensor<1,dim,Number> &src1,
-            const SymmetricTensor<2,dim,Number> &src2)
+template <int dim, typename Number, typename OtherNumber>
+Tensor<1,dim,typename ProductType<Number,OtherNumber>::type>
+operator * (const Tensor<1,dim,Number>               &src1,
+            const SymmetricTensor<2,dim,OtherNumber> &src2)
 {
   // this is easy for symmetric tensors:
   return src2 * src1;
 }
 
 
+
+/**
+ * The dot product (single contraction) for tensors: Return a tensor of rank
+ * $(\text{rank}_1 + \text{rank}_2 - 2)$ that is the contraction of the last
+ * index of a tensor @p src1 of rank @p rank_1 with the first index of a
+ * tensor @p src2 of rank @p rank_2:
+ * @f[
+ *   \text{result}_{i_1,\ldots,i_{r1},j_1,\ldots,j_{r2}}
+ *   = \sum_{k}
+ *     \text{left}_{i_1,\ldots,i_{r1}, k}
+ *     \text{right}_{k, j_1,\ldots,j_{r2}}
+ * @f]
+ *
+ * @note As one operand is a Tensor, the multiplication operator only performs a
+ * contraction over a single pair of indices. This is in contrast to the
+ * multiplication operator for SymmetricTensor, which does the double
+ * contraction.
+ *
+ * @note In case the contraction yields a tensor of rank 0 the scalar number
+ * is returned as an unwrapped number type.
```

Comment:
that can't happen here since the second argument has at least rank 2, so one contraction cannot yield a rank-0 object